### PR TITLE
fix: validate RGBA clipboard dimensions

### PR DESCRIPTION
**Pull request by FrederickStempfle**
## Summary

Validate peer-provided RGBA clipboard dimensions before constructing `arboard::ImageData`.

Previously, signed width and height fields were cast directly to `usize` and were not checked against the content length. A malformed clipboard message could therefore describe a very large image backed by only a few bytes, allowing downstream consumers to read beyond the supplied buffer.

## Changes

- reject non-positive dimensions
- use checked multiplication for `width * height * 4`
- require the computed RGBA length to exactly match the content length
- cover valid, truncated, negative, zero, and overflowing inputs

## Testing

`cargo test --lib --features linux-pkg-config rgba_tests::validates_dimensions_against_content_length`

The regression test passes.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved clipboard image validation for RGBA images by rejecting invalid or zero dimensions.
  * Ensures the clipboard payload size exactly matches the expected RGBA byte length before conversion.
  * Prevents malformed or oversized clipboard images from being converted incorrectly.
* **Tests**
  * Added non-Android unit tests covering valid cases, mismatched payload lengths, zero/negative dimensions, and overflow edge cases.

### CODE REVIEW

File: src/clipboard.rs [hunk 1]
```diff
@@ -770,6 +785,20 @@ mod proto {
...
+    use super::valid_rgba_dimensions;
+
+    #[test]
+    fn validates_dimensions_against_content_length() {
+        assert_eq!(valid_rgba_dimensions(1, 1, 4), Some((1, 1)));
+        assert_eq!(valid_rgba_dimensions(1, 1, 3), None);
+        assert_eq!(valid_rgba_dimensions(-1, 1, 4), None);
+        assert_eq!(valid_rgba_dimensions(0, 1, 0), None);
+        assert_eq!(valid_rgba_dimensions(i32::MAX, i32::MAX, 4), None);
+    }
```
[the remark this answers is not in the document]
Comment (FrederickStempfle) [reply src/clipboard.rs:801]: Fixed in 5f04e91. Added a 32-bit-gated case that reaches the checked multiplication overflow path. The targeted RGBA validation test passes with `--features linux-pkg-config`.
